### PR TITLE
Handle NextSeq's RunCompletionStatus.xml

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java [hunk 1]
```diff
@@ -36,10 +36,24 @@
...
+    try {
+      switch ((String) RUN_COMPLETION_STATUS_EXPRESSION.evaluate(document, XPathConstants.STRING)) {
+      case "CompletedAsPlanned":
+        return Optional.of(HealthType.Completed);
+      }
+    } catch (XPathExpressionException e) {
+      log.error("Failed to evaluate completion status", e);
+    }
+    return Optional.empty();
+  };
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java]: MINOR Remove this empty statement. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java [hunk 2]
```diff
@@ -36,10 +36,24 @@
 
   private static final Logger log = LoggerFactory.getLogger(DefaultIllumina.class);
 
+  private static final XPathExpression RUN_COMPLETION_STATUS_EXPRESSION = compileXPath("//CompletionStatus")[0];
+
   public static DefaultIllumina create(Builder builder, ObjectNode parameters) {
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java]: MAJOR Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java [hunk 3]
```diff
@@ -36,10 +36,24 @@
...
     return new DefaultIllumina(builder);
   }
 
+  private static Optional<HealthType> getHealth(Document document) {
+    try {
+      switch ((String) RUN_COMPLETION_STATUS_EXPRESSION.evaluate(document, XPathConstants.STRING)) {
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java]: CRITICAL Add a default case to this switch. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
MINOR Replace this "switch" statement by "if" statements to increase readability. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)

**Comment by sqwci** [comment]
SonarQube analysis reported 8 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DefaultIllumina.java#L68](https://github.com/TGAC/miso-lims/blob/e7643e286e7570d50df97a304f5443ecdfd03732/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L68): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [DefaultIllumina.java#L80](https://github.com/TGAC/miso-lims/blob/e7643e286e7570d50df97a304f5443ecdfd03732/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L80): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)
1. ![MAJOR][MAJOR] [DefaultPacBio.java#L177](https://github.com/TGAC/miso-lims/blob/e7643e286e7570d50df97a304f5443ecdfd03732/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java#L177): Extract the assignment out of this expression. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MINOR][MINOR] [DefaultIllumina.java#L93](https://github.com/TGAC/miso-lims/blob/e7643e286e7570d50df97a304f5443ecdfd03732/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L93): Refactor your code to get this URI from a customizable parameter. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1075)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DefaultIllumina.java#L68](https://github.com/TGAC/miso-lims/blob/9c9d5c1afd4652d99b9bb496e442ff88d8b7d06d/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L68): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [DefaultIllumina.java#L80](https://github.com/TGAC/miso-lims/blob/9c9d5c1afd4652d99b9bb496e442ff88d8b7d06d/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L80): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)
1. ![MAJOR][MAJOR] [DefaultPacBio.java#L177](https://github.com/TGAC/miso-lims/blob/9c9d5c1afd4652d99b9bb496e442ff88d8b7d06d/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java#L177): Extract the assignment out of this expression. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MINOR][MINOR] [DefaultIllumina.java#L93](https://github.com/TGAC/miso-lims/blob/9c9d5c1afd4652d99b9bb496e442ff88d8b7d06d/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L93): Refactor your code to get this URI from a customizable parameter. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1075)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java (hunk 3 above)
Comment (djcooke) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java]: Logging unknown values in the default case would be useful

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java [hunk 4]
```diff
@@ -36,10 +36,27 @@
...
+  private static final XPathExpression RUN_COMPLETION_STATUS_EXPRESSION = compileXPath("//CompletionStatus")[0];
+
   public static DefaultIllumina create(Builder builder, ObjectNode parameters) {
     return new DefaultIllumina(builder);
   }
 
+  private static Optional<HealthType> getHealth(Document document) {
+    try {
+      String status = (String) RUN_COMPLETION_STATUS_EXPRESSION.evaluate(document, XPathConstants.STRING);
+      switch (status) {
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java]: MINOR Replace this "switch" statement by "if" statements to increase readability. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java [hunk 5]
```diff
@@ -36,10 +36,27 @@
...
+      case "CompletedAsPlanned":
+        return Optional.of(HealthType.Completed);
+      default:
+        log.debug("New Illumina completion status found: " + status);
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java]: MAJOR Use the built-in formatting to construct this argument. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)

**Comment by sqwci** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DefaultIllumina.java#L71](https://github.com/TGAC/miso-lims/blob/5646bda62605e161075c1ce425b32c8d1ae0f3f4/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L71): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [DefaultIllumina.java#L83](https://github.com/TGAC/miso-lims/blob/5646bda62605e161075c1ce425b32c8d1ae0f3f4/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L83): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)
1. ![MAJOR][MAJOR] [DefaultPacBio.java#L177](https://github.com/TGAC/miso-lims/blob/5646bda62605e161075c1ce425b32c8d1ae0f3f4/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultPacBio.java#L177): Extract the assignment out of this expression. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
1. ![MINOR][MINOR] [DefaultIllumina.java#L96](https://github.com/TGAC/miso-lims/blob/5646bda62605e161075c1ce425b32c8d1ae0f3f4/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java#L96): Refactor your code to get this URI from a customizable parameter. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1075)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**